lib/proc/proc_select.cpp

```diff
@@ -4553,6 +4553,12 @@ grn_select_output_slice_label_v1(grn_ctx *ctx,
                                  grn_select_data *data,
                                  Slice *slice)
 {
+  if (grn_ctx_get_output_type(ctx) == GRN_CONTENT_APACHE_ARROW) {
+    ctx->impl->output.arrow_metadata_data_type = "slice";
+    ctx->impl->output.arrow_metadata_label = slice->label;
+    return;
+  }
+
   GRN_OUTPUT_STR(slice->label.value, slice->label.length);
```

Comment:
We should add a logical `grn_output_XXX()` function and use it instead of passing information via `ctx->impl->output`.

Reply:
fix: [869a3b6](https://github.com/groonga/groonga/pull/2399/commits/869a3b6e2e71c83230af0d5e8a6b9c1dc347b631) I added `grn_output_arrow_metadata()` function as a abstraction layer.

Reply:
fix: [53837a7](https://github.com/groonga/groonga/pull/2399/commits/53837a72db658aac6a345970d5ed7c22865d77c7) Sure. I've extracted the logic of setting the label.

Reply:
Hmm.

I feel that `grn_output_result_set_label()` -> `grn_output_result_set_open_metadata()` order is strange. Because `open` is a constructor name in Groonga. Calling other functions before `open()` is strange. (I feel that this mismatch introduces `ctx->impl->output.arrow_metadata_*`.)

Can we set label as a metadata? Can we use `grn_obj_format` for it?